### PR TITLE
Switch GPU test jobs to use install dir.

### PR DESCRIPTION
**Pull request by ScottTodd**
Progress on https://github.com/openxla/iree/issues/16203. I want to eventually extract this test job into a callable workflow file, but first seeing if I can drop the "build-dir" part of ci.yml.

ci-extra: test_a100

### CODE REVIEW

**Comment by ScottTodd** [comment]
Nice! This is about as fast as previous job now (and maybe even faster).

* Before: https://github.com/openxla/iree/actions/runs/7760512589/job/21168897867 (1m20s checkout + 1m15s download/extract)
* After: https://github.com/openxla/iree/actions/runs/7762451186/job/21173092277 (15s checkout + 40s download/extract + 1m45s build)

412 tests, down from 415

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -360,29 +376,37 @@ jobs:
...
+      - name: "Extracting install directory"
+        run: tar -xf "${INSTALL_DIR_ARCHIVE}"
+      - name: "Building tests"
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+              --env IREE_CPU_DISABLE \
+              --env IREE_VULKAN_DISABLE \
+              --env IREE_CUDA_DISABLE \
+              gcr.io/iree-oss/nvidia@sha256:892fefbdf90c93b407303adadfa87f22c0f1e84b7e819e69643c78fc5927c2ba \
+              ./build_tools/pkgci/build_tests_using_package.sh ${INSTALL_DIR}
```
Comment (ScottTodd) [inline .github/workflows/ci.yml:396]: between baseline and this PR, test_a100 is no longer running these tests:

* `iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test_cuda`
* `iree/tests/e2e/stablehlo_models/mnist_train_test/mnist_train_test_vulkan`
* `iree/tests/transform_dialect/cuda/double_mma_layout_analysis.mlir.test`
* `iree/tests/transform_dialect/cuda/mma_elemwise_layout_analysis.mlir.test`
* `iree/tests/transform_dialect/cuda/mma_reduction_layout_analysis.mlir.test`
* `iree/tests/transform_dialect/cuda/mma_using_layout_analysis.mlir.test`
* `iree/tests/transform_dialect/cuda/mma.mlir.test`

`iree/tests/transform_dialect/cuda` tests need this updated to use `IREE_HOST_BIN_DIR`:
https://github.com/openxla/iree/blob/10fd98b89616e7f1ae70b8c4036c2c715f1f5bc0/tests/transform_dialect/cuda/CMakeLists.txt#L13-L15

`mnist_train_test` I may let rot. We need a good way to plug Python tests in, and that test is .... not it right now :P

Comment (ScottTodd) [reply .github/workflows/ci.yml:396]: Pushed a commit that added the transform_dialect tests back. Still skipping mnist_train_test.

**Comment by ScottTodd** [comment]
Ping?

**Review by benvanik — APPROVED** [review]

**Review by pzread — APPROVED** [review]

**Pull request merged**